internal/encoding/parquet/parquet.go

```diff
@@ -111,13 +112,42 @@ func (i *iterator) Schema() typeof.Schema {
 	for _, c := range schema.Columns() {
 		t := c.Type()
 
-		if t, supported := typeof.FromParquet(t); supported {
-			result[c.Name()] = t
+		if t == nil {
+			p := getSchemaLogicalType(c.Element().GetLogicalType())
+
+			if t, supported := typeof.FromParquet(&p); supported {
+				result[c.Name()] = t
+			}
+		} else {
+			if t, supported := typeof.FromParquet(t); supported {
+				result[c.Name()] = t
+			}
 		}
 	}
 	return result
 }
 
+func getSchemaLogicalType(t *parquet.LogicalType) parquet.Type {
```

Comment:
Rename `getSchemaLogicalType` -> `parquetTypeOf(c *goparquet.Column)`, also add a comment. 
First thing, check if it's supported here so you can remove the code above:

```
func parquetTypeOf(c Column) parquet.Type {
 if t, supported := typeof.FromParquet(c.Type()); supported {
	return t
 }

 ... existing switch
}
```

Reply:
Actually, this function isnt really doing the actual check. It is only getting the logical type of the column if the physical type is not set since the logical type does not directly map to a physical type.